app/src/org/commcare/dalvik/activities/TemplatePrinterActivity.java

```diff
@@ -0,0 +1,310 @@
+
+import java.io.File;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import org.commcare.android.tasks.TemplatePrinterTask;
+import org.commcare.android.tasks.TemplatePrinterTask.PopulateListener;
+import org.commcare.android.util.TemplatePrinterUtils;
+import org.commcare.dalvik.R;
+import org.commcare.dalvik.application.CommCareApplication;
+import org.commcare.dalvik.preferences.CommCarePreferences;
+import org.javarosa.core.reference.InvalidReferenceException;
+import org.javarosa.core.reference.ReferenceManager;
+import org.javarosa.core.services.locale.Localization;
+
+import android.annotation.TargetApi;
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.SharedPreferences;
+import android.os.Build;
+import android.os.Bundle;
+import android.os.CancellationSignal;
+import android.os.ParcelFileDescriptor;
+import android.print.PageRange;
+import android.print.PrintAttributes;
+import android.print.PrintDocumentAdapter;
+import android.print.PrintJob;
+import android.print.PrintJobInfo;
+import android.print.PrintManager;
+import android.webkit.WebView;
+import android.webkit.WebViewClient;
+
+
+/**
+ * Intermediate activity which populates an HTML template with data and then prints it
+ * 
+ * @author Richard Lu
+ * @author amstone
+ */
+public class TemplatePrinterActivity extends Activity implements PopulateListener {
+
+    /**
+     * The path to the temp file location that is written to in TemplatePrinterTask, and then
+     * read back from in doHtmlPrint()
+     */
+    private String outputPath;
+
+    /**
+     * Unique name to use for the print job name
+     */
+    private static String jobName;
+
+    /**
+     * Used to hold an instance of the WebView object being printed, so that is it not garbage
+     * collected before the print job is created
+     */
+    private WebView mWebView;
+
+    private PrintJob printJob;
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_template_printer);
+
+        //Check to make sure we are targeting API 19 or above, which is where print is supported
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.KITKAT) {
+            showErrorDialog(Localization.get("print.not.supported"));
+        }
+
+        Bundle data = getIntent().getExtras();
+        //Check to make sure key-value data has been passed with the intent
+        if (data == null) {
+            showErrorDialog(Localization.get("no.print.data"));
+        }
+
+        this.outputPath = CommCareApplication._().getTempFilePath() + ".html";
+
+        //Check if a doc location is coming in from the Intent
+        //Will return a reference of format jr://... if it has been set
+        String path = data.getString("cc:print_template_reference");
```

Comment:
It doesn't disrupt the control flow of the code, so it will still reach this state.